src/Agent/NewRelic/Agent/Core/DistributedTracing/DistributedTracePayloadHandler.cs

```diff
@@ -74,6 +74,10 @@ public void InsertDistributedTraceHeaders<T>(IInternalTransaction transaction, T
                         setter(carrier, Constants.DistributedTracePayloadKeyAllLower, distributedTracePayload);
                     }
                 }
+                else
+                {
+                    transaction.SetSampled(_adaptiveSampler);
```

Comment:
Do we need to check if sampled is not already set first?

Reply:
Nope, `SetSampled()` only does anything if it hasn't been set yet